bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -253,6 +253,38 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
             .map(|((_, derivation_index), spk)| (*derivation_index, spk))
     }
 
+    /// Convenience method to get [`revealed_and_lookahead_spks_of_keychain`]
+    /// of all keychains.
+    ///
+    /// [`revealed_and_lookahead_spks_of_keychain`]: Self::revealed_and_lookahead_spks_of_keychain
+    pub fn revealed_and_lookahead_spks_of_all_keychains(
+        &self,
+    ) -> BTreeMap<K, impl Iterator<Item = (u32, &Script)> + Clone> {
+        self.keychains
+            .keys()
+            .map(|keychain| {
+                (
+                    keychain.clone(),
+                    self.revealed_and_lookahead_spks_of_keychain(keychain),
+                )
+            })
+            .collect()
+    }
+
+    /// Iterates over the script pubkeys revealed and the ones contained in
+    /// the lookahead under `keychain`.
+    pub fn revealed_and_lookahead_spks_of_keychain(
```

Comment:
we could also name it `stored_spks()` or something like that. the names are now getting a bit too verbose, 😅

We also should document somewhere the difference between all three kinds of spk lookups.